src/classes/HiFiCommunicator.ts

```diff
@@ -53,6 +67,58 @@ export enum HiFiUserDataStreamingScopes {
     All = "all"
 };
 
+export interface ConnectionRetryAndTimeoutConfig {
+  /**
+   * Whether or not to automatically retry initial connection attempts.
+   * When this is set to true, if the first attempt to connect to
+   * the High Fidelity servers fails, we will automatically retry the connection.
+   * While the connection is being attempted, the overall HiFiCommunicator state (as queried via
+   * `getConnectionState()`) will be "Connecting".
+   * After the desired amount of time has passed, if a connection has not been established, we will stop
+   * trying to reconnect and the connection state will change to 'Failed'.
+   * By default, connections are not retried. However, when we start a connection, we do always
+   * trigger a state change to HiFiConnectionStates.Connecting when we start the
+   * connection process, regardless of the setting of this value.
+   *
+   */
+  autoRetryInitialConnection?: boolean;
+  /**
+   * The total amount of time (in seconds) to keep retrying the initial
+   * connection before giving up completely. If `autoRetryInitialConnection`
+   * is set to `true`, this defaults to 60 seconds.
+   */
+  maxSecondsToSpendRetryingInitialConnection?: number;
```

Comment:
Style choice:
- A single maxSecondsToSpendRetryingInitialConnection, in which a number <= 0 will not retry, and no autoRetryInitialConnection boolean.
- A autoRetryInitialConnection boolean AND a maxSecondsToSpendRetryingInitialConnection number, as shown here, in which we document (and test at runtime) that the value supplied to maxSecondsToSpendRetryingInitialConnection is a number >= 0 and give a nice error message if it isn't.

My preference is the first, but I'm not willing to die on that hill. My reasoning is:
- It isn't any simpler or clearer for the user to have the two values, because with EITHER approach, the user still needs to read through and understand about negative numbers.
- While having two parameters allows a 60 second "default in case the user picks maxSecondsToSpendRetryingInitialConnection: true", there isn't anything magically righteous about 60 seconds. I'd rather have the application developer think about what kind of timeout is appropriate for their application.
- It takes more "space" to capture the state of the configuration, e.g., in logging the configuration to console or to HiFi for the purpose of examining connection failures.

Reply:
Let's discuss "in person" -- I also am not prepared to die on the hill, but it was easier for me just to move forward with what I had. Would be easy enough to change at this point though!